src/dataset.py

```diff
@@ -39,9 +39,20 @@ def license_lookup(license: str) -> str:
 
 def name_lookup(
     metadata: api_types.DatasetMetadata,
-    strict_search: bool = False,
+    entities: list[api_types.Entity],
```

Comment:
Unfortunately this kind of typing is only available in python 3.9 (for the time this project is locked to 3.8 I believe) - this should be `typing.List` instead of `list`

Reply:
Done in b8a487f